addons/delivery/models/stock_picking.py

```diff
@@ -153,7 +153,7 @@ def send_to_shipper(self):
         res = self.carrier_id.send_shipping(self)[0]
         if self.carrier_id.free_over and self.sale_id and self.sale_id._compute_amount_total_without_delivery() >= self.carrier_id.amount:
             res['exact_price'] = 0.0
-        self.carrier_price = res['exact_price']
+        self.carrier_price = res['exact_price'] * (1.0 + (self.carrier_id.margin / 100.0))
```

Comment:
In the send_shipping method call, the function to rate the shipment (for the given carrier) is called. What is called is not exactly the rate_shipment method of the carrier but the helper function that is used in rating the shipment. So basically, the carrier price calculated here is almost the same when calling rate_shipment except that in rate_shipment method, the margin is applied.